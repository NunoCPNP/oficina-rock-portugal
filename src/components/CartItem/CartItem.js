import Image from 'next/image'

import { Container, ImageContainer, DetailsContainer } from './CartItem.styles'

const CartItem = ({ item = {} }) => {
  return (
    <Container>
      <ImageContainer>
        <Image src={item.image} alt="" width="50" height="50" />
      </ImageContainer>
      <DetailsContainer>
        <div>
          <span>{item.type}</span>
          <div>{item.band}</div>
        </div>
      </DetailsContainer>
      <div>{item.quantitySelected}</div>
      <div>{item.size}</div>
      <div>{item.price * item.quantitySelected} â‚¬</div>
    </Container>
  )
}

export default CartItem
