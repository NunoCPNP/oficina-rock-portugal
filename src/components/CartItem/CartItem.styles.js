import styled from '@emotion/styled'

export const Container = styled.div`
  display: grid;
  grid-template-columns: auto 10rem auto auto auto;
  grid-gap: 2rem;
  align-items: center;
  margin-bottom: 1rem;
`

export const ImageContainer = styled.div`
  img {
    border-radius: 0.5rem;
  }
`

export const DetailsContainer = styled.div`
  span {
    font-size: var(--heading-5);
  }
`
