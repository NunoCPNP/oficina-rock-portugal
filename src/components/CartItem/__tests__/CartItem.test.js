import { render } from '@testing-library/react'

import CartItem from '../CartItem'

describe('<CartItem /> spec', () => {
  const props = {
    item: {
      image: '/image.jpg',
      type: 'T-Shirt',
      band: 'Ramones',
      quantitySelected: 2,
      size: 'M',
      price: 20,
    },
  }

  it('Should match snapshot', () => {
    const { container } = render(<CartItem {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should render item image', () => {
    const { getAllByRole } = render(<CartItem {...props} />)

    expect(getAllByRole('img')).toHaveLength(1)
  })

  it('Should display correct item information', () => {
    const { getByText } = render(<CartItem {...props} />)

    expect(getByText(props.item.type)).toBeInTheDocument()
    expect(getByText(props.item.band)).toBeInTheDocument()
    expect(getByText(props.item.quantitySelected)).toBeInTheDocument()
    expect(getByText(props.item.size)).toBeInTheDocument()
  })

  it('Should calculate and display price correctly', () => {
    const { getByText } = render(<CartItem {...props} />)

    expect(getByText(`${props.item.quantitySelected * props.item.price} â‚¬`)).toBeInTheDocument()
  })
})
