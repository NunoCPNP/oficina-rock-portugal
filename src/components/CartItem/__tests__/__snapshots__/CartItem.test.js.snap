// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> spec Should match snapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: auto 10rem auto auto auto;
  grid-gap: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
}

.emotion-1 img {
  border-radius: 0.5rem;
}

.emotion-2 span {
  font-size: var(--heading-5);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAiIGhlaWdodD0iNTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt=""
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div>
        <span>
          T-Shirt
        </span>
        <div>
          Ramones
        </div>
      </div>
    </div>
    <div>
      2
    </div>
    <div>
      M
    </div>
    <div>
      40
       â‚¬
    </div>
  </div>
</div>
`;
